lightning/src/ln/data_persister.rs

```diff
@@ -0,0 +1,221 @@
+	/// is a risk because previous channel states are toxic, so it's important
+	/// that whatever channel state is persisted is kept up-to-date.
+	///
+	/// If an implementer chooses to persist the updates only, they need to make
+	/// sure that all the updates are applied to the ChannelMonitors *before* the
+	/// set of channel monitors is given to the ChainMonitor at startup time
+	/// (e.g., all monitors returned by [`load_channel_data`] must be up-to-date).
+	/// If full ChannelMonitors are persisted, then there is no need to persist
+	/// individual updates.
+	///
+	/// In the case where one or more backups fail, it's likely safe to only
+	/// require that the majority of backups succeed and agree on the latest
+	/// [`update_id`] to succeed this method.
+	///
+	/// See [`ChannelMonitor::write_for_disk`] for writing out a ChannelMonitor
+	/// and [`ChannelMonitorUpdate::write`] for writing out an update.
+	///
+	/// [`load_channel_data`]: trait.ChannelDataPersister.html#tymethod.load_channel_data
+	/// [`update_id`]: struct.ChannelMonitorUpdate.html#structfield.update_id
+	/// [`ChannelMonitor::write_for_disk`]: struct.ChannelMonitor.html#method.write_for_disk
+	/// [`ChannelMonitorUpdate::write`]: struct.ChannelMonitorUpdate.html#method.write
+	fn update_channel_data(&self, id: OutPoint, update: &ChannelMonitorUpdate, data: &ChannelMonitor<Self::Keys>) -> Result<(), ChannelMonitorUpdateErr>;
+
+	/// Load the data for all channels. Generally only called on startup. You must
+	/// ensure that the ChannelMonitors returned are on the latest [`update_id`],
+	/// with all of the updates given in [`update_channel_data`] applied. Otherwise,
+	/// there is a risk of broadcasting a revoked transaction and losing money.
+	///
+	/// See [`ChannelMonitor::read`] for deserializing a ChannelMonitor.
+	///
+	/// [`update_id`]: struct.ChannelMonitorUpdate.html#structfield.update_id
+	/// [`update_channel_data`]: trait.ChannelDataPersister.html#tymethod.update_channel_data
+	/// [`ChannelMonitor::read`]: trait.Readable.html
+	fn load_channel_data(&self) -> Result<HashMap<OutPoint, ChannelMonitor<Self::Keys>>, ChannelMonitorUpdateErr>;
+}
+
+/// LinuxPersister can persist channel data on disk on Linux machines, where
+/// each channel's data is stored in a file named after its funding outpoint.
+pub struct LinuxPersister<ChanSigner: ChannelKeys + Readable + Writeable> {
+	path_to_channel_data: String,
+	phantom: PhantomData<ChanSigner>, // TODO: is there a way around this?
+}
+
+impl<ChanSigner: ChannelKeys + Readable + Writeable> LinuxPersister<ChanSigner> {
+	/// Initialize a new LinuxPersister and set the path to the individual channels'
+	/// files.
+	pub fn new(path_to_channel_data: String) -> Self {
+		return Self {
+			path_to_channel_data,
+			phantom: PhantomData,
+		}
+	}
+
+	fn get_full_filepath(&self, funding_txo: OutPoint) -> String {
+		format!("{}/{}_{}", self.path_to_channel_data, funding_txo.txid.to_hex(), funding_txo.index)
```

Comment:
Consider using `std::path` module.

https://doc.rust-lang.org/std/path/index.html

Reply:
I think I'm missing something but what specific benefit were you thinking? It makes the logic for formulating the `tmp` file a teeny bit more complicated because you can't `clone()` paths..

Reply:
My thinking is it would take care of platform-specific separator characters. You can still return a `String` of course.

Reply:
This brings up a related point: this is called `LinuxPersister`, but it's actually cross-platform as far as as I can tell.  i.e. it should work fine on MacOS or BSD.  So if you use `std::path`, it might work well on Windows too.

Reply:
Updated to use `std::path` and tested on Windows and MacOS.